changes in file (#1)
Bravinsimiyu/yeah: new.md

### Introduction

> this is a new API links allow you to acquire content from other websites and display it wherever you like. In this tutorial, you will learn how to create a javaScript joke generator using API links which is an excellent project for novice developers who may want to rapidly make plenty of humorous jokes.

Reviewer: ```suggestion
this is a new API link that allows you to acquire content from other websites and display it wherever you like. In this tutorial, you will learn how to create a javaScript joke generator using API links which is an excellent project for novice developers who may want to rapidly make plenty of humorous jokes.
```

#### Prerequesites

- This article will only be useful if the following is in place before reading it:
  Basic HTML, CSS, and JavaScript expertise are required for each of the following:

#### Objectives

> - By the end of this tutorial learners should be able to:

Reviewer: changes needed

  1. Design a basic `Html` container to contain content.
  2. Connect API links to get data from other web pages.

### Table of content

- [Introduction](#introduction)
  - [Prerequesites](#prerequesites)
  - [Objectives](#objectives)
- [Table of content](#table-of-content)
- [About Example](#about-example)
- [Basic structure of Joke Generator](#basic-structure-of-joke-generator)
- [Adding a title](#adding-a-title)
- [Set up Joke Viewing Zone](#set-up-joke-viewing-zone)
- [Set up Generate Joke Button](#set-up-generate-joke-button)
- [Activating The Button](#activating-the-button)
- [Conclusion](#conclusion)
  - [Further reading](#further-reading)

### About Example

For the comedy API, we're going to employ HTML, CSS, and javascript as our building blocks. We'll start by adding a header to a box on a webpage. A viewing area and a button will be set up next, which will create our amusing content when pressed.
The following files will be available:

- index.html
- style.css
- app.js

This is the file structure.

![output-file-arrangement](engineering-education/random-joke-generator-a-complete-guide/output-file-arrangement.png)

### Basic structure of Joke Generator

> The `background-color` of our basic box will be `#0C0C0C` and it will be `550px` wide. We'll use `(border-radius:5px)` of `5px` to round the edges of the box. Refer to the following example.

Reviewer: ```suggestion
The `background-color` of our basic box will be `#0C0C0C` and it will be `550px` wide. We'll use `(border-radius:5px)` of `5px` to round the edges of the box. Refer to the following example.d
```

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Random Joke Generator</title>
  </head>
  <body>
    <!-- box -->
    <div class="box"></div>
  </body>
</html>
```

The box will be on the top center position of our page by setting its `position` to `absolute`, on the `left` of our page, we will have `50%` as well as the `top` and lastly, we will `transform` and `translate` our box from the top and left as `-50%`.

```css
.box {
  width: 550px;
  padding: 40px 50px;
  background-color: #0c0c0c;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border-radius: 5px;
}
```

The box should resemble the following picture in appearance.

> ![output-box](engineering-education/random-joke-generator-a-complete-guide/output-box.png)

Reviewer: ```suggestion
![output-box](engineering-education/random-joke-generator-a-complete-guide/output-box.png)x
```
